data/users.js

```diff
@@ -1,8 +1,112 @@
+ * @returns 
+ */
+async function getMentorList(userId){
+    validate.checkArgLength(arguments, 1);
+    userId = validate.convertID(userId);
+    let usersDB = await getUsersCollection();
+
+    let userObj = await usersDB.findOne({_id: userId});
+    if (userObj == null) throw `Error: Could not find ${userId}`;
+
+    return userObj.mentorRelationships;
+}
+
+/**
+ * Returns the given user's menteeRelationships list
+ * @param {Object|String} userId 
+ * @returns 
+ */
+async function getMenteeList(userId){
+    validate.checkArgLength(arguments, 1);
+    userId = validate.convertID(userId);
+    let usersDB = await getUsersCollection();
+
+    let userObj = await usersDB.findOne({_id: userId});
+    if (userObj == null) throw `Error: Could not find ${userId}`;
+
+    return userObj.menteeRelationships;
+}
+
+/**
+ * Returns both the given user's mentor and mentee relationships in a single list
+ * @param {Object|String} userId 
+ * @returns 
+ */
+async function getUserRelationships(userId){
+    validate.checkArgLength(arguments, 1);
+    userId = validate.convertID(userId);
+    let usersDB = await getUsersCollection();
+
+    let userObj = await usersDB.findOne({_id: userId});
+
+    let relationships = [];
+    relationships.push(...userObj.mentorRelationships);
+    relationships.push(...userObj.menteeRelationships);
+
+    return relationships;
+}   
+
+/**
+ * Updates mentor and mentee's relationship lists according to what role they play in the current relationhsip
+ * Returns the updated user object corresponding to the given Id
+ * @param {Object|String} userId 
+ * @param {Object} relationshipObj 
+ * @returns 
+ */
+async function updateUserRelationships(userId, relationshipObj){
+    validate.checkArgLength(arguments, 2);
+    let mentorId = validate.convertID(relationshipObj.mentor);
+    let menteeId = validate.convertID(relationshipObj.mentee);
+    
+    let usersDB = await getUsersCollection();
+    //let userObj = await usersDB.findOne({_id: userId});
+    
+    let mentor = await usersDB.findOne({_id: mentorId});
+    let mentee = await usersDB.findOne({_id: menteeId});
+    if (mentor == null || mentee == null) throw `Error: Could not find user ${mentor} or ${mentee}`;
+    mentor.menteeRelationships.push(relationshipObj._id);
+    mentee.mentorRelationships.push(relationshipObj._id);
+
+    let setMentorObj = {
+        $set: {
+            menteeRelationships: mentor.menteeRelationships
+        }
+    };
+
+    let updatedMentor = await usersDB.findOneAndUpdate({_id: mentorId}, setMentorObj);
+
+    let setMenteeObj = {
+        $set: {
+            mentorRelationships: mentee.mentorRelationships
+        }
+    };
+
+    let updatedMentee = await usersDB.findOneAndUpdate({_id: menteeId}, setMenteeObj);
+
+    if(updatedMentor == null || updatedMentee == null) throw `Failed to update user relationships mentee: ${updatedMentee} ; mentor: ${updatedMentor}`;
+    
+    userId = validate.convertID(userId);
```

Comment:
I guess we need the validation at the start of the function:
- UserId is a valid String that can be converted to ObjectId
- UserId should be equal to either Mentor ID or Mentee ID of the relation (Unless you want to do this validation for all the Relationship routes in middleware)

Reply:
Because any operation related to a particular Relation can **ONLY** be done by either Mentor or Mentee of the relation.